pkg/certs/cert-inspection/certgraphanalysis/collector.go

```diff
@@ -69,90 +76,274 @@ func platformSecrets(obj *corev1.Secret) bool {
 
 func gatherFilteredCerts(ctx context.Context, kubeClient kubernetes.Interface, acceptConfigMap configMapFilterFunc, acceptSecret secretFilterFunc, options certGenerationOptionList) (*certgraphapi.PKIList, error) {
 	inClusterResourceData := &certgraphapi.PerInClusterResourceData{}
-	certs := []*certgraphapi.CertKeyPair{}
-	caBundles := []*certgraphapi.CertificateAuthorityBundle{}
 	errs := []error{}
 
-	// TODO here is the point where need to collect data like node names and IPs that need to be replaced
-	//  this will be something like options.Discovery(kubeClient, configClient).
+	caBundles, err := gatherConfigMaps(ctx, kubeClient, acceptConfigMap, options, inClusterResourceData)
+	if err != nil {
+		errs = append(errs, err)
+	}
 
-	configMapList, err := kubeClient.CoreV1().ConfigMaps("").List(ctx, metav1.ListOptions{})
-	switch {
-	case err != nil:
+	certs, err := gatherSecrets(ctx, kubeClient, acceptSecret, options, inClusterResourceData)
+	if err != nil {
 		errs = append(errs, err)
-	default:
-		for _, configMap := range configMapList.Items {
-			if !acceptConfigMap(&configMap) {
-				continue
-			}
-			if options.rejectConfigMap(&configMap) {
-				continue
-			}
-			name, namespace := options.rewriteConfigMap(&configMap)
-			details, err := InspectConfigMap(&configMap, name, namespace)
-			if err != nil {
-				errs = append(errs, err)
-				continue
-			}
-			if details == nil {
-				continue
-			}
-			caBundles = append(caBundles, details)
-
-			inClusterResourceData.CertificateAuthorityBundles = append(inClusterResourceData.CertificateAuthorityBundles,
-				certgraphapi.PKIRegistryInClusterCABundle{
-					ConfigMapLocation: certgraphapi.InClusterConfigMapLocation{
-						Namespace: namespace,
-						Name:      name,
-					},
-					CABundleInfo: certgraphapi.PKIRegistryCertificateAuthorityInfo{
-						OwningJiraComponent: configMap.Annotations[annotations.OpenShiftComponent],
-						Description:         configMap.Annotations[annotations.OpenShiftDescription],
-					},
-				})
+	}
+
+	pkiList := PKIListFromParts(ctx, inClusterResourceData, certs, caBundles)
+	return pkiList, errors.NewAggregate(errs)
+}
+
+func GatherCertsFromDisk(ctx context.Context, kubeClient kubernetes.Interface, dir string, options ...*certGenerationOptions) (*certgraphapi.PKIList, error) {
```

Comment:
I'm guessing the plan is to create a command in `openshift-tests` or in `oc`.  If that's the case, create the `*Options` struct with the `Validate` and `Run` methods for standard embedding.

I'd expect to also see this library also contain the code (likely also in a different `*Options` struct) to 
1. select nodes
2. force schedule pods to each node
3. exfiltrate the result of the command

Running the collection as an init container, then a sleep as the main container will allow you to `oc copy` (embedded) the result out.

Reply:
I'd really prefer to avoid running new containers - its always not as reliable as "fetch data from already running container"